pal/src/map/virtual.cpp

```diff
@@ -534,6 +537,19 @@ static PCMI VIRTUALFindRegionInformation( IN UINT_PTR address )
 
     pEntry = pVirtualMemory;
 
+    if (pVirtualMemoryLastFound && pVirtualMemoryLastFound->startBoundary <= address)
+    {
+        pEntry = pVirtualMemoryLastFound;
```

Comment:
Sorry, I don't understand the intent here- correct me if I'm wrong but this is how I understand the current code:
- pVirtualMemory is a list of allocated "regions", stored in increasing order of startAddress (hence the check on line 555)
- Your idea (which makes sense) is to have a 1 element MRU cache that you check first before walking the list- that is, check if the address is within the most recently used region

Is that right? If so, why are you checking for exact start instead of `address <= pEntry->startBoundary + pEntry->memSize`? It seems like the current condition is much more restrictive than it needs to be?

Also, minor nit but I personally find the assignment to pEntry redundant- I would rather check pVirtualMemoryLastFound directly instead of assigning to pEntry first but that's just me :)

Reply:
memSize not needed. This is to help us if the address we are looking is further from the previous search/find point on the linked list or not. Only condition we need for that is startBoundary <= address.

Reply:
Can you explain why you're doing `pEntry->startBoundary == address` instead of within a region? For example, consider this case- your list has `A (startBoundary: 1000), B (startBoundary: 2000). `pVirtualMemoryLastFound` is set to `A`. You call `VIRTUALFindRegionInformation` with 1500- this should be a cache hit, but you'd miss the cache in this case, no? What did I misunderstand?

Reply:
Not sure I understood. Let's say we've been looping from the beginning. Wouldn't we loop until the pVirtualMemoryLastFound? and loop further or not? so, we set the loop start point here and do the rest similarly?